docs/source/GenerateExamples.py

```diff
@@ -0,0 +1,98 @@
+import os
```

Comment:
Can you explain why this script is needed? 
I think you can do a much simpler setup for sphinx gallery and specify those in the conf.py. https://sphinx-gallery.github.io/stable/configuration.html